sympy/functions/elementary/integers.py

```diff
@@ -144,6 +144,12 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
         arg = self.args[0]
         arg0 = arg.subs(x, 0)
         r = self.subs(x, 0)
+        if arg0 is S.NaN:
+            limit = arg.limit(x, 0, dir='-' if re(cdir).is_negative else '+')
```

Comment:
A global name like `limit` is not a good choice for a local variable. If the limit exists, it means that `arg` can be extended continuously to `x = 0` and the limit is its value at 0. So this could be `arg0` redefined.